data/src/main/java/pl/edu/agh/model/LocalDateTimePeriod.java

```diff
@@ -0,0 +1,66 @@
+package pl.edu.agh.model;
+
+import java.time.DateTimeException;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import javax.persistence.Column;
+import javax.persistence.Embeddable;
+import lombok.AllArgsConstructor;
+import lombok.Getter;
+import lombok.Setter;
+
+@Setter
+@Getter
+@AllArgsConstructor
+@Embeddable
+public class LocalDateTimePeriod {
+
+    @Column(nullable = false)
+    private LocalDateTime start;
+
+    @Column(nullable = false)
+    private LocalDateTime end;
+
+    public LocalDateTimePeriod() {
+        this.start = LocalDateTime.now();
+        this.end = this.start;
+    }
+
+    public void setStart(LocalDateTime start) {
+        if (start.isAfter(this.end)) {
+            throw new DateTimeException("Start date cannot be after the end date.");
+        }
+        this.start = start;
+    }
+
+    public void setEnd(LocalDateTime end) {
+        if (end.isBefore(this.start)) {
+            throw new DateTimeException("Start date cannot be after the end date.");
+        }
+        this.end = end;
+    }
+
+    public void set(LocalDateTime start, LocalDateTime end) {
+        if (start.isBefore(end)) {
+            throw new DateTimeException("Start date cannot be after the end date.");
```

Comment:
Tu jest chyba troszkę bug ;)
Ja to mam poprawione w jednym commicie razem z testami do tej klasy, to żeby Ci tu nie robić zamieszania, to jak skończysz ogarniać resztę, to go dorzucę go na brancha

Reply:
no fakt, na odwrot powinno byc